MineInAbyss/src/main/java/com/derongan/minecraft/mineinabyss/Ascension/Effect/Effects/AbstractAscensionEffect.java

```diff
@@ -19,8 +26,12 @@
 
     @Override
     public void applyEffect(Player player, int ticks) {
-        durationRemaining -= ticks;
-        applyEffect(player);
+        for (int IterationsScheduled = 0; IterationsScheduled != iterations; IterationsScheduled++) {
```

Comment:
Minor: Variables should be camelCase.

Reply:
Also, did you consider doing this differently?

Right now all the scheduled hits are pre-scheduled, which makes cancelling them harder if a player drinks a potion of immunity or whatever we have in the game for this.

Instead, you could have each task schedule the next task recursively. 

int numIterations;

```
Bukit....runTaskLater(....()->{
    numIterations -= 1;
    if(numIterations > 0){
    applyEffect
    }
});
```

This doesn't exactly work probably due to complaints about the counter not being final, but there are ways around that (wrap it in an object for example)